core/xmipp_metadata_program.cpp

```diff
@@ -170,7 +167,7 @@ void XmippMetadataProgram::readParams()
     track_origin = track_origin || checkParam("--track_origin");
     keep_input_columns = keep_input_columns || checkParam("--keep_input_columns");
 
-    MetaData * md = new MetaData;
+    MetaData * md = new MetaDataVec();
```

Comment:
is this not memory leak?

Reply:
It is not memory leak. Use of `std::unique_ptr` would help to understand why... `setup` method takes ownership of the pointer and maintains it.